src/KubernetesClient/Kubernetes.ConfigInit.cs

```diff
@@ -179,8 +180,11 @@ private void SetCredentials(KubernetesClientConfiguration config, HttpClientHand
 
                 // add all your extra certificate chain
                 chain.ChainPolicy.ExtraStore.Add(caCert);
-                chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
-                var isValid = chain.Build((X509Certificate2) certificate);
+                chain.Build((X509Certificate2) certificate);
+
+                var rootCert = chain.ChainElements[chain.ChainElements.Count - 1].Certificate;
+                var isValid = rootCert.RawData.SequenceEqual(caCert.RawData);

```

Comment:
If you want to test for a custom root you'd leave the check as it was, and then add  `isValid = isValid && rootCert.RawData.SequenceEqual(caCert.RawData);`

As it is, this change would start accepting expired and revoked certificates (and possibly certificates where the signatures didn't check out).

Reply:
Thank you!